### ISSUE TITLE
Plans to support multiclass classification?

### ISSUE THREAD

**Issue by rquintino** [Microsoft/LightGBM#43]
or through a 1 vs all classifier?

**Comment by wxchan** [comment]
I am trying to work on this feature, hope what I have done is right. 

**Current version is [here](https://github.com/wxchan/LightGBM/tree/dev). I might need some help on fixing details and making test.** 

You can run examples in examples/multiclass_classification and use compare.py to compare the result between lightgbm and xgboost. You can replace the data with yours and see if it works.

You need to add `objective = multiclass` and `metric = multi_logloss(or multi_error)` in train.conf. You also need to specify `num_class = ` (must >=2) in both train.conf and predict.conf. The output of prediction is probability distribution on each classes, so num_class float numbers per line. Validation sets probably not work yet (don't set early_stopping_round now, it will write more models unexpectedly), I will fix it out later, but it won't affect anything when `objective != multiclass`.

**Comment by chivee** [comment]
@rquintino , please check the latest code for the Multiclass Classfication example
